test/e2e/framework/framework.go

```diff
@@ -445,9 +449,43 @@ func (f *Framework) CreateNamespace(ctx context.Context, baseName string, labels
 	// fail to create serviceAccount in it.
 	f.AddNamespacesToDelete(ns)
 
+	if TestContext.DockerConfigFile != "" {
+		secret, err := f.CreateSecretFromDockerConfig(ctx, ns.Name)
+		ExpectNoError(err)
+
+		serviceAccountClient := f.ClientSet.CoreV1().ServiceAccounts(ns.Name)
+		serviceAccountConfig := v1svc.ServiceAccount(serviceAccountName, ns.Name)
+		serviceAccountConfig.ImagePullSecrets = append(serviceAccountConfig.ImagePullSecrets, v1svc.LocalObjectReferenceApplyConfiguration{Name: &secret.Name})
+
+		svc, err := serviceAccountClient.Apply(ctx, serviceAccountConfig, metav1.ApplyOptions{FieldManager: "application/apply-patch"})
+		if err != nil {
+			return ns, fmt.Errorf("[%s] Failed to patch imagePullSecrets to service account [%s]: %v", svc.Name, secret.Name, err)
+		}
+
+	}
+
 	return ns, err
 }
 
+// CreateSecretFromDockerConfig creates a secret using the private image registry credentials.
+// The credentials are provided by --docker-config-file flag.
+func (f *Framework) CreateSecretFromDockerConfig(ctx context.Context, namespace string) (*v1.Secret, error) {
+	contents, err := os.ReadFile(TestContext.DockerConfigFile)
+	ExpectNoError(err)
```

Comment:
Please pick one way of handling errors in this function: either use an error return value or assertions, but not both (here `ExpectNoError`, below return).

Whatever you choose, add additional strings that explain what the error is. Adding "reading Docker config file" will help the user understand what the file is when the file name itself doesn't make it clear.

Reply:
Thanks for the input, made the changes.